### ISSUE TITLE
SSR + <script> causes re-execution

### ISSUE THREAD

**Issue by wmertens** [nfl/react-helmet#357]
When I try to put a `<script>` tag in `<Helmet>` and do SSR, it goes into the head just fine, but then when the app renders, it runs the script again :-/

I looked at the source some to figure out what's going on, but no idea.

In fact, it seems that every time the `<Helmet>` is rendered, it does the same, even when the performance flamegraph shows that it's an update and not an unmount/mount.

Perhaps `<script>` tags should be treated more carefully and checked for equivalence before applying?

In the meantime I'll work around it by only rendering the `<scripts>` on the server.

**Comment by Vaibhav** [comment]
Check this out https://github.com/NYTimes/react-helmet-async

**Comment by tmbtech** [comment]
Yeah we have noticed that too, we usually don't load external scripts with helmet. The current API doesn't support a way to know when the script has been loaded and as you stated has the potential to get loaded multiple times.

This is a bug and will need to be addressed.

**Comment by Kadrian** [comment]
I'm seeing this too, using `gatsby` and `gatsby-plugin-react-helmet`, my script tags get re-executed.

**Comment by JimmyBcn** [comment]
Does anybody has found a way to fix scripts in `<Helmet>` being being re-executed using `gatsby-plugin-react-helmet`?

**Comment by wmertens** [comment]
@JimmyBcn you'll have to use a global to see if the script was executed already
